### PR TITLE
Control task

### PR DESCRIPTION
**Pull request by NicolaiFransen**
The things needed for closed-loop operation has been added as comments. 
I have considered if the executeControl() should have a return value so the scheduler knows what mode of operation is selected. What do you think?

### CODE REVIEW

File: Software/Development/SemesterProject/App/controlTask.c [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+#include "controlTask.h"
+
+/*
+ * This function will check the state of the open-loop/closed-loop switch,
+ * and from that execute the desired operation mode.
+ */
+void executeControl(void)
+{
+    if (isOpenLoopControlSelected())
```
Comment (murgui) [inline Software/Development/SemesterProject/App/controlTask.c]: as discussed, I think that the system manager state should be checked here. It seems a simpler system.

So the idea would be:
if (sysMgrState = Running)  {    do_control()}
else error_reaction();

error_reaction()
{
    set_duties_to_0();
    set_driver_enable_to_0();
    restart_control_vars();//here or downstream, just think about it
    //more error reactions?
}

File: Software/Development/SemesterProject/App/controlTask.c [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+void disableDutyCycles(void)
+{
+    setAllDuties(0);
+    // Consider to disable the PWM enable output pin as well
+}
+
+/*
+ * This function will return the state of the open-loop/closed-loop switch.
+ */
+int isOpenLoopControlSelected(void)
```
Comment (murgui) [inline Software/Development/SemesterProject/App/controlTask.c:53]: nice interface

**Review by murgui — CHANGES_REQUESTED** [review]
only a couple of comments

File: Software/Development/SemesterProject/App/controlTask.c (hunk 1 above)
Comment (NicolaiFransen) [reply Software/Development/SemesterProject/App/controlTask.c]: This have been changed now

File: Software/Development/SemesterProject/App/controlTask.c [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ * This function will check the state of the open-loop/closed-loop switch,
+ * and from that execute the desired operation mode.
+ */
+void executeControl(void)
+{
+    if (readSystemState() == RUNNING)
+    {
+        if (isOpenLoopControlSelected())
+            // Call open-loop function
+            setFrequencyFromPot();
```
Comment (murgui) [inline Software/Development/SemesterProject/App/controlTask.c]: this function name changed

**Review by murgui — CHANGES_REQUESTED** [review]
almost

**Review by murgui — APPROVED** [review]
Ship it!

**Review by AitorTeran — APPROVED** [review]

**Pull request merged**